### PR TITLE
Add converter tool for toc.md

### PR DESCRIPTION
**Pull request by qinezh**

### CODE REVIEW

File: tools/TocMarkdownConverter/Program.cs [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+                return 1;
+            }
+
+            var tocYml = args.Length > 1 ? args[1] : null;
+            try
+            {
+                tocYml = TocConverter.Convert(tocMarkdown, tocYml);
+                Console.WriteLine($"Successfully convert {tocMarkdown} to {tocYml}.");
+
+                return 1;
```
Comment (superyyrrzz) [inline tools/TocMarkdownConverter/Program.cs]: >1 [](start = 23, length = 1)

0?

File: tools/TocMarkdownConverter/Program.cs [hunk 2]
```diff
@@ -0,0 +1,48 @@
+﻿using System;
+using System.IO;
```
Comment (superyyrrzz) [inline tools/TocMarkdownConverter/Program.cs]: copyright header, move into namespace

File: tools/TocMarkdownConverter/Program.cs [hunk 3]
```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace TocMarkdownConverter
```
Comment (vicancy) [inline tools/TocMarkdownConverter/Program.cs]: naming: TocConverter? md->yml or yml->md can be options

File: tools/TocMarkdownConverter/TocConverter.cs [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+            {
+                throw new FileNotFoundException($"{tocMarkdownFilePath} can't be found.");
+            }
+
+            if (string.IsNullOrEmpty(tocYmlPath))
+            {
+                var tocName = Path.GetFileNameWithoutExtension(tocMarkdownFilePath);
+                var tocDir = Path.GetDirectoryName(tocMarkdownFilePath);
+
+                tocYmlPath = Path.Combine(tocDir, tocName + YmlExtension);
```
Comment (vicancy) [inline tools/TocMarkdownConverter/TocConverter.cs]: use Path.ChangeExtension

File: tools/TocMarkdownConverter/TocConverter.cs [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+            }
+
+            ConvertCore(tocMarkdownFilePath, tocYmlPath);
+
+            return tocYmlPath;
+        }
+
+        private static void ConvertCore(string tocMarkdownFilePath, string tocYmlPath)
```
Comment (vicancy) [inline tools/TocMarkdownConverter/TocConverter.cs]: `tocYmlPath` not needed

File: tools/TocMarkdownConverter/TocMarkdownConverter.csproj [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+    <TargetFramework>net46</TargetFramework>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <ProjectReference Include="..\..\src\Microsoft.DocAsCode.Build.TableOfContents\Microsoft.DocAsCode.Build.TableOfContents.csproj" />
+    <ProjectReference Include="..\..\src\Microsoft.DocAsCode.Common\Microsoft.DocAsCode.Common.csproj" />
+    <ProjectReference Include="..\..\src\Microsoft.DocAsCode.DataContracts.Common\Microsoft.DocAsCode.DataContracts.Common.csproj" />
+  </ItemGroup>
+
+  <ItemGroup Condition=" '$(TargetFramework)' == 'net46' OR '$(TargetFramework)' == 'net452'">
```
Comment (vicancy) [inline tools/TocMarkdownConverter/TocMarkdownConverter.csproj]: `Condition` not needed

File: tools/TocMarkdownConverter/TocMarkdownConverter.csproj [hunk 7]
```diff
@@ -0,0 +1,20 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+  <Import Project="..\..\src\Shared\base.props" />
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>net46</TargetFramework>
```
Comment (vicancy) [inline tools/TocMarkdownConverter/TocMarkdownConverter.csproj]: no need `TargetFramework`

File: tools/TocMarkdownConverter/TocMarkdownConverter.csproj (hunk 6 above)
Comment (qinezh) [inline tools/TocMarkdownConverter/TocMarkdownConverter.csproj]: Copied from [MergeSourceInfo.csproj](https://github.com/dotnet/docfx/blob/dev/tools/MergeSourceInfo/MergeSourceInfo.csproj#L14-L18) ...

File: tools/TocMarkdownConverter/TocConverter.cs (hunk 5 above)
Comment (qinezh) [inline tools/TocMarkdownConverter/TocConverter.cs]: Why not needed

File: tools/TocConverter/Program.cs [hunk 8]
```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace TocConverter
```
Comment (hellosnow) [inline tools/TocConverter/Program.cs]: >TocConverter [](start = 10, length = 12)

prepend namepspace name "Microsoft.DocAsCode"?

**Pull request merged**